src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -241,6 +246,13 @@ public ProduceMetrics(final Metrics metrics, final Map<String, String> tags) {
       _produceDelay.add(new MetricName("produce-delay-ms-avg", METRIC_GROUP_NAME, "The average delay in ms for produce request", tags), new Avg());
       _produceDelay.add(new MetricName("produce-delay-ms-max", METRIC_GROUP_NAME, "The maximum delay in ms for produce request", tags), new Max());
 
+      // There are 2 extra buckets use for values smaller than 0.0 or larger than max, respectively.
+      int bucketNum = _latencyPercentileMaxMs / _latencyPercentileGranularityMs + 2;
+      int sizeInBytes = 4 * bucketNum;
+      _produceDelay.add(new Percentiles(sizeInBytes, _latencyPercentileMaxMs, Percentiles.BucketSizing.CONSTANT,
```

Comment:
Using fixed width histograms  seems like a poor choice for capturing the tails of distributions.  Kafka  does not have anything else to choose from?  Exponential decay?  Dynamic histograms?

Reply:
Yeah Kafka does not have other choice.